### PR TITLE
Desktop: networking

### PR DESCRIPTION
**Pull request by akimd**
### Proposed changes

Update the documentation of Docker for Desktop networking.  Steal the Mac doc to start the Windows one.

I am unable to make snapshots of the Windows proxy setting pane, so I can't write the section ATM.  I will address this later (unless someone beats me :).

Related to https://github.com/docker/desktop-experience-backlog/issues/37.

### CODE REVIEW

File: docker-for-windows/networking.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+## Known limitations, use cases, and workarounds
+
+Following is a summary of current limitations on the Docker for {{Arch}}
+networking stack, along with some ideas for workarounds.
+
+### There is no docker0 bridge on {{Arch}}
+
+Because of the way networking is implemented in Docker for {{Arch}}, you cannot
+see a `docker0` interface in macOS.  This interface is actually within the
```
Comment (gbarr01) [inline docker-for-windows/networking.md]: macOS?

File: docker-for-windows/networking.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+However, it works with Windows Containers.
+
+### Use cases and workarounds
+
+There are two scenarios that the above limitations affect:
+
+#### I want to connect from a container to a service on the host
+
+The host has a changing IP address (or none if you have no network access). From
+18.03 onwards our recommendation is to connect to the special Mac-only DNS name
```
Comment (gbarr01) [inline docker-for-windows/networking.md]: Mac-only?

**Comment by akimd** [comment]
Hi @gbarr01.  Sorry for the mistakes, I tried to rush before finishing my day, never a good idea...

Since then I have found that I could address the missing documentation about proxies in Networking by harmonizing the documentations of for-Mac and for-Windows.  So there's a lot of movement here, but it's basically just that: moving things around, to reach a clearer structure (hopefully).

**Comment by akimd** [comment]
Hi @gbarr01!  I believe this is ready to be installed.

**Comment by akimd** [comment]
@gtardif PTAL.

File: docker-for-mac/index.md [hunk 3]
```diff
@@ -159,28 +159,50 @@ to decrease it, lower the number.
...
+TERM=xterm
+HOME=/root
+HTTP_PROXY=http://proxy.example.com:3128
+http_proxy=http://proxy.example.com:3128
+no_proxy=*.local, 169.254/16
+```
+
+You can see from the above output that the `HTTP_PROXY`, `http_proxy`, and
+`no_proxy` environment variables are set. When your proxy configuration changes,
+Docker restarts automatically to pick up the new settings. If you have
```
Comment (gtardif) [inline docker-for-mac/index.md]: I thought VPNKit was able to catch thse changes without restarting, can you double-check it restarts when changing the system proxies @akimd ?

File: docker-for-mac/networking.md [hunk 4]
```diff
@@ -5,74 +5,58 @@ redirect_from:
...
 ## Features
 
 ### VPN Passthrough
 
-Docker for Mac's networking can work when attached to a VPN. To do this, Docker
-for Mac intercepts traffic from the `HyperKit` and injects it into macOS as if
-it originated from the Docker application.
+Docker for {{Arch}}'s networking can work when attached to a VPN. To do this,
+Docker for {{Arch}} intercepts traffic from the containers and injects it into
+Windows as if it originated from the Docker application.
```
Comment (gtardif) [inline docker-for-mac/networking.md]: {{Arch}} instead of Windows here

File: docker-for-windows/networking.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+Docker for {{Arch}} provides several networking features to make it easier to
+use.
+
+## Features
+
+### VPN Passthrough
+
+Docker for {{Arch}}'s networking can work when attached to a VPN. To do this,
+Docker for {{Arch}} intercepts traffic from the containers and injects it into
+Windows as if it originated from the Docker application.
```
Comment (gtardif) [inline docker-for-windows/networking.md]: {{Arch}} instead of Windows here

File: docker-for-mac/index.md (hunk 3 above)
Comment (akimd) [reply docker-for-mac/index.md]: I have tried to change my system Proxy, and not restart Docker for Mac.  Then I have:
```
$ docker run -it alpine env
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=83bdc8002344
TERM=xterm
HOME=/root
```
Then I have restarted Docker for Mac, and... same result.

So I don't know what is going wrong here (maybe just the doc), but this is probably that should be dealt with in another issue.  Let's apply this one anyway (the piece of doc is not new, it's just moved).

File: docker-for-mac/index.md [hunk 6]
```diff
@@ -94,7 +94,7 @@ Choose ![whale menu](/docker-for-mac/images/whale-x.png){: .inline} -> **Prefere
 
 ![Docker context menu](images/menu/d4m-menu-prefs.png){:width="250px"}
 
-### General tab
+### General
```
Comment (gbarr01) [inline docker-for-mac/index.md]: editing for style? ;)

Comment (akimd) [reply docker-for-mac/index.md]: Yep.  As written in the commit message, there are many gratuitous differences between both docs, and I tried to harmonize them, choosing what seemed most appropriate.  And in many occurrences, the shortest was the best (IMHO).

File: docker-for-windows/index.md [hunk 7]
```diff
@@ -237,86 +237,6 @@ and make sure `docker` commands are working properly.
...
-
-    ```ps
-    Notepad $PROFILE
-    ```
-
-Open a new PowerShell session. Now, when you press tab after typing the first
-few letters, Docker commands such as `start`, `stop`, `run`, and their options,
-along with container and image names should now auto-complete.
-
 ## Docker Settings
```
[the remark this answers is not in the document]
Comment (akimd) [reply docker-for-windows/index.md]: On Mac the menu item is called "Preferences...", on Windows it is called "Settings...".  I guess these are the standard wordings for each platform.

File: docker-for-mac/networking.md [hunk 8]
```diff
@@ -5,74 +5,58 @@ redirect_from:
...
-```
-
-You can see from the above output that the `HTTP_PROXY`, `http_proxy`, and
-`no_proxy` environment variables are set. When your proxy configuration changes,
-Docker restarts automatically to pick up the new settings. If you have
-containers that you wish to keep running across restarts, you should consider
-using [restart policies](/engine/reference/run/#restart-policies-restart).
+See [Proxies](index.md:Proxies).
 
 ## Known limitations, use cases, and workarounds
```
[the remark this answers is not in the document]
Comment (akimd) [reply docker-for-mac/networking.md]: Let's discuss this elsewhere, once the PR merged.  It's already really big.

**Comment by akimd** [comment]
Hi @gbarr01.  I have fix the "searching" issue.   I have also updated the Troubleshooting part.

I can hardly touch anything until my PRs are merged, further work depends on this.

Cheers!

**Pull request merged**